File: Source/Mobile/src/services/auth-api.ts
import axiosClient from "@/lib/axios-interceptor";

export interface LoginRequestDto {
  accountName: string;
  password?: string;
}

export interface LoginResponseDto {
  message: string;
  employeeId: number;
  employeeName: string | null;
}

export interface UserProfileDto {
  employeeId: number;
  employeeCode: string | null;
  employeeAccount: string;
  employeeFullName: string | null;
  titleId: number | null;
  employeeBirthDate: string | null;
  maritalStatus: number | null;
  gender: number | null;
  employeeMobile: string | null;
  employeePhone: string | null;
  employeeEmail: string | null;
  employeeAddress: string | null;
  userCreated: number | null;
  userModified: number | null;
  dateCreated: string | null;
  dateModified: string | null;
  hireDate: string | null;
  status: number | null;
  departmentId: number | null;
  others: string | null;
  defaultPage: string | null;
  employeeImageIcon: string | null;
  employeeType: number | null;
  userRoles: string | null;
  workTypeId: number | null;
}

export const authApi = {
  login: (payload: LoginRequestDto, tenantCode: string) => {
    return axiosClient.post<any, LoginResponseDto>("/Auth/login", payload, {
      withCredentials: true,
      headers: {
        "X-Tenant-Code": tenantCode,
      },
    });
  },

  getMe: () => {
    return axiosClient.get<any, UserProfileDto>("/Auth/me");
  },

  logout: async () => {
    await axiosClient.post<any, void>(
      "/Auth/logout",
      {},
      {
        withCredentials: true,
      },
    );
  },
};

File: Source/flutter/lib/services/auth_api.dart
import '../models/auth_dto.dart';
import 'api_client.dart';

class AuthApi {
  static final ApiClient _apiClient = ApiClient();

  /// Đăng nhập vào hệ thống
  /// Mirrors: authApi.login(payload: LoginRequestDto, tenantCode: string)
  static Future<LoginResponseDto> login(
    String tenantCode, {
    required String accountName,
    required String password,
  }) async {
    final payload = LoginRequestDto(
      accountName: accountName,
      password: password,
    );

    final response = await _apiClient.post(
      '/Auth/login',
      body: payload.toJson(),
      headers: {
        'X-Tenant-Code': tenantCode,
      },
    );

    return LoginResponseDto.fromJson(response as Map<String, dynamic>);
  }

  /// Lấy thông tin tài khoản đang đăng nhập
  /// Mirrors: authApi.getMe()
  static Future<UserProfileDto> getMe() async {
    final response = await _apiClient.get('/Auth/me');
    return UserProfileDto.fromJson(response as Map<String, dynamic>);
  }

  /// Đăng xuất khỏi hệ thống
  /// Mirrors: authApi.logout()
  static Future<String> logout() async {
    String message = "Đăng xuất thành công!";
    try {
      final response = await _apiClient.post('/Auth/logout');
      if (response is Map<String, dynamic> && response['message'] != null) {
        message = response['message'].toString();
      }
    } finally {
      _apiClient.clearCookies();
    }
    return message;
  }
}
